### PR TITLE
Give LambdaExpr some useful constructors.

### LINKED ISSUE #1997: Why is it valid to create a `LambdaExpr` with `isEnclosingParameters=false` with non-single parameter list?
Hey,

```java
new LambdaExpr(new NodeList<>(), ..., false)
```
would result in illegal Java code, i.e. (`-> ...`, without params).
Is there maybe a way to ignore the `isEnclosingParameters` param in case it doesn't make sense?
Also, for multiple parameters it should also be ignored, i.e. the `isEnclosingParameters` parameter only makes sense for parameter lists with single values.

Would it make sense to only include this option for a constructor that accepts a single parameter and make it `true` for `NodeList`s?

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1997

### CODE REVIEW

**Comment by matozoid** [comment]
I guess there could be a check to see if you're passing a single parameter to the multi-parameter constructor and not add () then... but most utility constructors are very straightforward.

@paplorinc - what do you think?

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java [hunk 1]
```diff
@@ -27,15 +31,45 @@ private void assertRange(String startToken, String endToken, Node node) {
...
+    @Test
+    void oneParameterAndExpressionUtilityConstructor() {
+        LambdaExpr expr = new LambdaExpr(new Parameter(new UnknownType(), "a"), parseExpression("5"));
+        assertEquals("a -> 5", expr.toString());
+    }
+
+    @Test
+    void oneParameterAndStatementUtilityConstructor() {
+        LambdaExpr expr = new LambdaExpr(new Parameter(new UnknownType(), "a"), parseStatement("return 5;"));
+        assertEquals("a -> return 5;", expr.toString());
```
Comment (l0rinc) [inline javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java]: I don't think this is valid Java code...

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java [hunk 2]
```diff
@@ -27,15 +31,45 @@ private void assertRange(String startToken, String endToken, Node node) {
...
+    @Test
+    void multipleParametersAndExpressionUtilityConstructor() {
+        LambdaExpr expr = new LambdaExpr(new NodeList<>(new Parameter(new UnknownType(), "a"), new Parameter(new UnknownType(), "b")), parseExpression("5"));
+        assertEquals("(a, b) -> 5", expr.toString());
+    }
+
+    @Test
+    void multipleParametersAndStatementUtilityConstructor() {
+        LambdaExpr expr = new LambdaExpr(new NodeList<>(new Parameter(new UnknownType(), "a"), new Parameter(new UnknownType(), "b")), parseStatement("return 5;"));
+        assertEquals("(a, b) -> return 5;", expr.toString());
```
Comment (l0rinc) [inline javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java]: same

**Review by l0rinc — CHANGES_REQUESTED** [review]
Thanks

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java (hunk 1 above)
Comment (matozoid) [reply javaparser-core-testing/src/test/java/com/github/javaparser/ast/expr/LambdaExprTest.java]: Ooooooops!

**Pull request merged**